docs/content/docs/build/tokenproviders.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Tutorial: Writing a TokenProvider with an Azure Function"
+menuPosition: 3
+editor: sdeshpande3
+---
+
+The token provider is responsible for creating and signing tokens that the `@fluidframework/azure-client` uses to make
+requests to the Azure Fluid Relay service. There are two ways by which you can generate the token:
+
+- `InsecureTokenProvider`
+
+- Implement your own `TokenProvider` class for fetching the token and a backend for token generation
+
+Fluid provides an `InsecureTokenProvider` that accepts your tenant secret, then locally generates and returns a signed token. This type of token provider is only useful for development purpose. For production scenarios, you must use a secure token provider.
+
+## Implementing your own TokenProvider class
+The `TokenProvider` class can be created by extending the `ITokenProvider` interface, thereby not exposing the tenant key secret in the client-side bundle code. While the tenant ID, orderer and storage URLs are fine to be on client-side code, the tenant key itself should not be exposed as it would allow malicious users to gain access to your tenant. The `TokenProvider` class would fetch the token from your very own backend service, thus providing a secure way for token resolution.
```

Comment:
```suggestion
## Implementing your own TokenProvider class

The `TokenProvider` class can be created by extending the `ITokenProvider` interface, thereby not exposing the tenant key secret in the client-side bundle code. While the tenant ID, orderer and storage URLs are fine to be on client-side code, the tenant key itself should not be exposed as it would allow malicious users to gain access to your tenant. The `TokenProvider` class would fetch the token from your very own backend service, thus providing a secure way for token resolution.
```